faunadb-java-dsl/src/main/java/com/faunadb/client/types/Value.java

```diff
@@ -529,4 +529,35 @@ public String apply(Byte input) {
       return format("BytesV(%s)", str);
     }
   }
+
+  /**
+   * Represents a query value in the FaunaDB query language.
+   *
+   * @see <a href="https://fauna.com/documentation/queries#values-special_types">FaunaDB Special Types</a>
+   */
+  @JsonDeserialize(using = JsonDeserializer.None.class) // Disables generic value deserializer for query values
+  public static final class QueryV extends Value {
+
+    private Map<String, Object> lambda;
+
+    private QueryV(@JsonProperty("@query") Map<String, Object> lambda) {
+      this.lambda = lambda;
+    }
+
+    @Override
+    @JsonProperty("@query")
+    protected Object toJson() {
```

Comment:
minor: you could still declare this as `protected Map<String, Object> toJson()`. Makes the contract less vague.